vendor/kcfinder/core/bootstrap.php

```diff
@@ -108,6 +108,42 @@ function checkLSSession()
     {
         define("BASEPATH", realpath($sLimesurveyFolder . "/../vendor/yiisoft/yii/framework") . "/");
     }
+
+    if (!defined('YII_DEBUG')) {
+        if (file_exists(APPPATH.'config'.DIRECTORY_SEPARATOR.'config.php')) {
```

Comment:
config is already read here : https://github.com/LimeSurvey/LimeSurvey/blob/47c5aa59c3163da201af9b7269e8a20920b25813/vendor/kcfinder/core/bootstrap.php#L114

(see [internal](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/config/internal.php#L13))

then action must be done after in my option.

Reply:
The changes the you suggest would require to move `$config = require_once(APPPATH . 'config/internal.php'); ` and some other includes to the top. 

To me, the position of those includes, taken for the index.php file, is set on purpose.
Wouldn't change that now and do that code rorg right now, but leave it to some other code reorg. The code reorg to the config files I believe is indeed needed, but wouldn't tackle it now.

Yes, code is repeating itself form index.php. But we are not creating a new entry of duplicated code, just fixing old duplicates. Not the best, but I don't think this suggested code reorg is priority right now.

@olleharstedt Thoughts?

Reply:
You answer on both comments her .

Why not

```
    $config = require_once(APPPATH . 'config/internal.php');
    /* Set debug system here */
    require_once APPPATH . '../vendor/autoload.php';
    require_once BASEPATH . 'yii.php';
    require_once APPPATH . 'core/LSYii_Application.php';
    Yii::$enableIncludePath = false;
```

Why moving debug system after loading break something ?

Reply:
> Why moving debug system after loading break something ?

Just feel the order is like that and can't be updated that easily. Not sure if  want to mess with it right now for this bug fix.
It is like a pandora box.

For example: 
- config/internal.php doesn't rely on anything from autoload?
- I have just spotted config/internal.php relies on YII_DEBUG being set already.
- I have just spotted config/internal.php relies on CMap, so Yii should already be loaded.
...

So, I think experimenting with the order doesn't fit in this ticket.